base/applications/rapps/unattended.cpp

```diff
@@ -254,14 +254,23 @@ ParseCmdAndExecute(LPWSTR lpCmdLine, BOOL bIsFirstLaunch, int nCmdShow)
         if ((!hMutex) || (GetLastError() == ERROR_ALREADY_EXISTS))
         {
             /* If already started, find its window */
```

Comment:
The `hMutex` check at old line 270 should be moved, or at least copied, here.
Moreover, I assume no mutex should mean to fail: otherwise, how could `ERROR_ALREADY_EXISTS` happen then?
Note: See #3200 for a similar case.

Reply:
I figured the original code was like this because the mutex is not required for the program to function and I just kept this behavior.

Reply:
> When another process creates the mutex first

Yeah, but what if the first process failed to create the mutex, for some reason?

--

To confirm, can multiple instances of RApps work fine simultaneously?

While here, note:
https://learn.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-createmutexexa
`If you are using a named mutex to limit your application to a single instance, a malicious user can create this mutex before you do and prevent your application from starting. To prevent this situation, create a randomly named mutex and store the name so that it can only be obtained by an authorized user. Alternatively, you can use a file for this purpose. To limit your application to one instance per user, create a locked file in the user's profile directory.`

Reply:
RApps works with multiple instances for the most part. The critical part is probably the database update but that happens before the mutex anyway (this PR does not change that). The database update should probably have its own mutex.

![screenshot](https://github.com/reactos/reactos/assets/138629473/296c097d-dc70-4945-89f1-c8bea247aea2)


You cannot protect yourself from a process running as the same user at the same (or higher) privilege level. Named kernel objects are per session unless you use the Global prefix. The only scenario where a lockfile is better than a mutex is when the malicious program is running at Low Integrity Level (UAC feature).